src/KeyVault/KeyVault/Models/VaultManagementClient.cs

```diff
@@ -86,61 +140,436 @@ public PSKeyVault CreateNewVault(VaultCreationOrUpdateParameters parameters, IMi
             if (string.IsNullOrWhiteSpace(parameters.Location))
                 throw new ArgumentNullException("parameters.Location");
 
-            var properties = new VaultProperties();
+            parameters.Name = GenerateDeploymentName(parameters);
+            Deployment deployment = CreateBasicDeployment(parameters, parameters.DeploymentMode, parameters.DeploymentDebugLogLevel, networkRuleSet);
+            parameters.Location = null;
 
-            if (parameters.CreateMode != CreateMode.Recover)
+            this.BeginDeployment(parameters, deployment, cmdlet);
+            var dep = ProvisionDeploymentStatus(parameters, deployment, cmdlet).ToPSDeployment(resourceGroupName: parameters.ResourceGroupName);
+            
+            return GetVault(parameters.Name, parameters.ResourceGroupName);
+        }
+        private void WriteDeploymentProgress(VaultCreationOrUpdateParameters parameters, DeploymentOperationErrorInfo deploymentOperationError)
+        {
+            var result = ListDeploymentOperations(parameters.ResourceGroupName, parameters.Name);
+            var newOperations = GetNewOperations(operations, result);
+            operations.AddRange(newOperations);
+
+            foreach (DeploymentOperation operation in newOperations)
             {
-                if (string.IsNullOrWhiteSpace(parameters.SkuFamilyName))
-                    throw new ArgumentNullException("parameters.SkuFamilyName");
-                if (parameters.TenantId == Guid.Empty)
-                    throw new ArgumentException("parameters.TenantId");
-                if (!string.IsNullOrWhiteSpace(parameters.SkuName))
+                if (operation.Properties.ProvisioningState == ProvisioningState.Failed.ToString())
                 {
-                    if (Enum.TryParse(parameters.SkuName, true, out SkuName skuName))
-                    {
-                        properties.Sku = new Sku(skuName);
-                    }
-                    else
+                    deploymentOperationError.ProcessError(operation);
+                }
+            }
+        }
+        public DeploymentExtended ProvisionDeploymentStatus(VaultCreationOrUpdateParameters parameters, Deployment deployment, KeyVaultManagementCmdletBase cmdlet = null)
+        {
+            operations = new List<DeploymentOperation>();
+            ProvisioningState[] status = new ProvisioningState[] { ProvisioningState.Canceled, ProvisioningState.Succeeded, ProvisioningState.Failed };
+            var getDeploymentFunc = this.GetDeploymentAction(parameters);
+            var deploymentOperationError = new DeploymentOperationErrorInfo();
+            Action writeProgressAction = () => this.WriteDeploymentProgress(parameters, deploymentOperationError);
+
+            int progressBarTimeSpan = 25;
+            int step = 5;
+            int phaseOne = 10;
+            var downloadStatus = new ProgressStatus(0, progressBarTimeSpan);
```

Comment:
```suggestion
            int progressBarTimeSpan = 25;
            int step = 5;
            int phaseOne = 10;
            var downloadStatus = new ProgressStatus(0, progressBarTimeSpan);
```
Move to the static properties